packages/ecmascript/src/Module.ts

```diff
@@ -16,11 +16,64 @@ const GLOBAL = System.getContext() || (function () {
 	return this;
 }).call(null);
 
+/**
+ * Function to handle errors when loading/importing an action or module.
+ * @param {string | Error} err - error (message)
+ */
+export type ModuleErrorHandler = (error: string | Error) => void;
+
+/** Predefined {@link ModuleErrorHandler error handler options} for use in {@link Module.setModuleErrorHandler} */
+enum DefaultModuleErrorHandlers {
+	/** [Default] Creates a System ERROR level log entry for the error. */
+	SYS_ERROR,
+	/** Creates a System WARNING level log entry for the error. */
+	SYS_WARN,
+	/** Creates a System INFO level log entry for the error. */
+	SYS_INFO,
+	/** Creates a System DEBUG level log entry for the error. */
+	SYS_DEBUG,
+	/** Ignores the error. */
+	SILENT,
+	/** Rethtows the error without handling it. */
+	THROW_ERROR
+}
+
 export interface ModuleConstructor extends Function {
+	/**
+	 * @param {string[]} specifiers - listed names of elements (actions/modules) to import from a Module.
+	 * At least 1 is required.
+	 * Also accepts "default" and "*" as specifiers for the default exports/full contents of the module.
+	 * The module can be specified via chained call to from(modulePath)
+	 * @returns {ModuleImport}
+	 */
 	import(...specifiers: string[]): ModuleImport;
-	export(path: string): ModuleExport;
+	/**
+	 * Initiates the export of an element.
+	 * @returns {ModuleExport} - builder instance.
+	 */
+	export(): ModuleExport;
+	/**
+	 * Loads a module (namespace) by its name. Error handling (as defined in
+	 * {@link ModuleConstructor.setModuleErrorHandler}) is invoked if:
+	 * - the path is missing or invalid
+	 * - the System cannot load a module for the given path.
+	 * @param {string} name - module name:
+	 * - may consist of one or more words (comprising of numbers, latin lower/upercase letters, '-' or '_'),
+	 * separated by '.'
+	 * - may not start or end with '.'
+	 * @returns {any} - action or module. Null if the path is null/undefined
+	 * 					or if System cannot find a module with the given path.
+	 */
 	load(path: string): ModuleDescriptor;
-	require(path: string): any;
+
+	/**
+	 * Changes the active {@link ModuleErrorHandler} used in the import and load operations
+	 * @param {DefaultModuleErrorHandlers | ModuleErrorHandler} eh - can be:
+	 * - a custom module {@link ModuleErrorHandler error handler function}
+	 * - one of the predefined {@link DefaultModuleErrorHandlers} options
+	 * Default is {@link DefaultModuleErrorHandlers.SYS_ERROR}
+	 */
+	setModuleErrorHandler(eh: DefaultModuleErrorHandlers | ModuleErrorHandler): void;
```

Comment:
Do we really want to be accepting a `string` or a `function`? It's breaking the "S" in SOLID.